tests/Unit/Helpers/DataStructures/Tensor/Expressions/EvaluateRank3TestHelpers.hpp

```diff
@@ -185,33 +183,31 @@ void test_evaluate_rank_3_impl(const TensorIndexA& tensorindex_a,
 /// TensorExpression
 /// \tparam ValenceC the valence of the third index used on the RHS of the
 /// TensorExpression
-/// \param tensorindex_a the first TensorIndex used on the RHS of the
+/// \tparam TensorIndexA the first TensorIndex used on the RHS of the
 /// TensorExpression, e.g. `ti_a`
-/// \param tensorindex_b the second TensorIndex used on the RHS of the
+/// \tparam TensorIndexB the second TensorIndex used on the RHS of the
 /// TensorExpression, e.g. `ti_B`
-/// \param tensorindex_c the third TensorIndex used on the RHS of the
+/// \tparam TensorIndexC the third TensorIndex used on the RHS of the
 /// TensorExpression, e.g. `ti_c`
 template <typename DataType,
           template <size_t, UpLo, typename> class TensorIndexTypeA,
           template <size_t, UpLo, typename> class TensorIndexTypeB,
           template <size_t, UpLo, typename> class TensorIndexTypeC,
-          UpLo ValenceA, UpLo ValenceB, UpLo ValenceC, typename TensorIndexA,
-          typename TensorIndexB, typename TensorIndexC>
-void test_evaluate_rank_3_no_symmetry(
-    const TensorIndexA& tensorindex_a, const TensorIndexB& tensorindex_b,
-    const TensorIndexC& tensorindex_c) noexcept {
+          UpLo ValenceA, UpLo ValenceB, UpLo ValenceC, auto& TensorIndexA,
+          auto& TensorIndexB, auto& TensorIndexC>
+void test_evaluate_rank_3_no_symmetry() noexcept {
```

Comment:
As I was updating these signatures to work with the new convention, I realized that these test functions should probably eventually be refactored now that `TensorIndex` tracks index type and valence. I don't think these functions really need to be called with anything besides the `DataType` and the `TensorIndex`s, since the valence and index type can be extracted from the latter. This applies to all of the other functions like this, including the other ranks and the similar tests in the `ComputeRhsTensorIndexRankNTestHelpers.hpp` files.

I didn't implement these potential refactor changes in this PR just to keep the changes focused on removing the 
`TensorIndex` aliases (and also because I was anticipating the changes for this PR to be a quicker fix than it was, since I'm working on the products now), but let me know if you'd like me to do something about it in this PR.

Reply:
I think doing that in a separate PR is the right call :) Much easier to review 2 small PRs than one big one